### PR TITLE
Fix empty contact code string errors

### LINKED ISSUE #8492: Null is not an object (evaluating `r.substring`) when trying to start new chat with empty contact code
### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Bug

*Summary*: error is shown at attempt to start 1-1 chat with empty or erased value. In both cases, error appears (in 1 case - `Null is not an object (evaluating "r.substring") `, in 2 -   `Null is not an object (evaluating "a.substring") `)

#### Expected behavior
1) validation error 
2) no error

#### Actual behavior
1) `Null is not an object (evaluating `r.substring`)`
2) `Null is not an object (evaluating `a.substring`)`
IMG_6338
image

### Reproduction

**Scenario 1:**
- Open Status
- Tap on + > Start new chat
- Set focus on field with contact code  and submit empty value 

**Scenario 2:**
- Open Status
- Tap on + > Start new chat
- Set focus on field with contact code, input some value
- Erase value

### Additional Information

[comment]: # (Please do your best to fill this out.)
- Status version: nightly 25/06/2019
[comment]: # (Android or iOS.)
- Operating System: Android, iOS

#### Logs
[status_logs1.zip](https://github.com/status-im/status-react/files/3325355/status_logs1.zip)

### PR DESCRIPTION
**Pull request by vitvly**
Fix #8492

### CODE REVIEW

File: src/status_im/ui/screens/add_new/new_chat/events.cljs [hunk 1]
```diff
@@ -20,7 +20,7 @@
      (merge {:db (assoc db
                         :contacts/new-identity       new-identity
                         :contacts/new-identity-error (db/validate-pub-key db new-identity))}
-            (when-not is-public-key?
+            (when-not (or is-public-key? (string/blank? new-identity))
```
Comment (flexsurfer) [inline src/status_im/ui/screens/add_new/new_chat/events.cljs]: could you elaborate please?

Comment (vitvly) [reply src/status_im/ui/screens/add_new/new_chat/events.cljs]: Empty string is not a public key.

Comment (vitvly) [reply src/status_im/ui/screens/add_new/new_chat/events.cljs]: And so ens name check fails.

**Review by flexsurfer — APPROVED** [review]

**Comment by churik** [comment]
failed e2e are not related to PR.
@siphiuel 
The only one issue that I noticed: if you paste smth with `.` in the end (i.e. `example.com.`) in "Enter contact code" - the error `Null is not an object (evaluating "n.substring") ` will be shown.
But it is a really small low-prio issue in comparison with https://github.com/status-im/status-react/issues/8492 - so if it is not trivial to fix, let's merge this PR.

**Comment by vitvly** [comment]
@churik thanks for testing, added a fix for trailing dot and also for two consecutive dots in the middle (strings like "a..a" would trigger the same error message)

**Pull request merged**